bugshaker/src/main/java/com/github/stkent/bugshaker/BugShaker.java

```diff
@@ -44,277 +44,260 @@
-                    BugShaker.class.getClassLoader());
-
-            logger.d("Detected that embedding app includes Google Maps as a dependency.");
-
-            return new MapScreenshotProvider(application, logger);
-        } catch (final ClassNotFoundException e) {
-            logger.d("Detected that embedding app does not include Google Maps as a dependency.");
-
-            return new BasicScreenShotProvider(application, logger);
-        }
-    }
-
-    private DialogProvider getAlertDialogProvider() {
-        if (alertDialogType == AlertDialogType.APP_COMPAT) {
-            try {
-                Class.forName(
-                        "android.support.v7.app.AlertDialog",
-                        false,
-                        BugShaker.class.getClassLoader());
-
-                logger.d("Using AppCompat dialogs as requested.");
-
-                return new AppCompatDialogProvider();
-            } catch (final ClassNotFoundException e) {
-                logger.e("AppCompat dialogs requested, but class not found.");
-                logger.e("Falling back to native dialogs.");
-
-                return new NativeDialogProvider();
-            }
-        } else {
-            logger.d("Using native dialogs as requested.");
-
-            return new NativeDialogProvider();
-        }
-    }
+	private static final String RECONFIGURATION_EXCEPTION_MESSAGE =
+		"Configuration must be completed before calling assemble or start";
+
+	private static BugShaker sharedInstance;
+
+	public static final String PREFS_NAME = "name";
+	public static final String PREFS_NAME2 = "name2";
+
+	public static final String PREFS_KEY = "key";
+	public static final String PREFS_KEY2 = "key2";
+
+
+	private final Application application;
+	private FeedbackEmailFlowManager feedbackEmailFlowManager;
+	private Logger logger;
+	public static String[] emailAddresses;
+	public static String emailSubjectLine;
+	private AlertDialogType alertDialogType = AlertDialogType.NATIVE;
+	private boolean loggingEnabled = false;
+
+	// Instance configuration state:
+	private boolean assembled = false;
+	private boolean startAttempted = false;
+
+
+	private final SimpleActivityLifecycleCallback simpleActivityLifecycleCallback
+		= new SimpleActivityLifecycleCallback() {
+
+		@Override
+		public void onActivityResumed(final Activity activity) {
+			feedbackEmailFlowManager.onActivityResumed(activity);
+		}
+
+		@Override
+		public void onActivityStopped(final Activity activity) {
+			feedbackEmailFlowManager.onActivityStopped();
+		}
+	};
+
+	/**
+	 * @param application the embedding application
+	 * @return the singleton <code>BugShaker</code> instance
+	 */
+	public static BugShaker get(@NonNull final Application application) {
+		synchronized (BugShaker.class) {
+			if (sharedInstance == null) {
+				sharedInstance = new BugShaker(application);
+			}
+		}
+
+		return sharedInstance;
+	}
+
+	public void save(Context context, String text) {
+		SharedPreferences settings;
```

Comment:
Can we move this out to SharedPreferenceUtil? Ideally Bugshaker should not know how to save things to a SharedPreference.